Model/Client/AbstractClient.php

```diff
@@ -175,6 +194,40 @@ protected function getHeaders(): array
         return $header;
     }
 
+    /**
+     * Get information about Magento version executing the request.
+     *
+     * @return array
+     */
+    protected function getReferrerData(): array
+    {
+        return [
+            'type' => 'magento',
+            'version' => $this->metadata->getVersion()
+        ];
+    }
+
+    /**
+     * Create request to Airwallex.
+     *
+     * @param Client $client
+     * @return ResponseInterface
+     * @throws GuzzleException
+     */
+    protected function createRequest(Client $client): ResponseInterface
+    {
+        $method = $this->getMethod();
+        $options = $this->getRequestOptions();
+
+        if ($method === 'POST') {
+            $this->params['request_id'] = $this->identityService->generateId();
+            $this->params['referrer_data'] = $this->getReferrerData();
```

Comment:
Should we only add the referrer data on POST requests? Or do we want to add it to any request?

Reply:
Hi @laurisdedumets 

After discussion with our team, we decided that the extension version of Magento (e.g 1.1.3 ) is a more important data point for us. Therefore, we would like to have it revised to pass the extension version instead of Magento version. Are you able to revise it for us? I will update the ticket as well.

Apologies for the inconvenience caused and let us know if there are any issues.

Cheers
Snow